contents/tutorials/group-page-machine-flags.md

```diff
@@ -0,0 +1,159 @@
+
+## Targeting pages
+
+Another target you might want is pages. For example, to ensure components are consistent across a set of pages, test a new pricing page, or customize a page's experience based on how many times it has been viewed.
+
+### Payloads
+
+You can use [feature flag payloads](/docs/feature-flags/payloads) to customize the experience depending on the page. For example, if you want a button to change colors depending on the page URL. Payloads lets you set multiple values on one feature flag which you can evaluate and use in your app.
+
+To do this, create your flag with a payload such as:
+
+```json
+{
+  "signup": "red",
+  "login": "blue",
+  "home": "green"
+}
+```
+
+In your app, get feature flag payloads, write logic to check the page URL, and use a combination of the page URL and flag payload to adjust your button. For a Next.js component, this would look like this:
+
+```js-web
+'use client'
+import { useFeatureFlagPayload } from 'posthog-js/react'
+import { usePathname } from 'next/navigation'
+
+export default function Button() {
+
+  const payload = useFeatureFlagPayload('button-colors')
+  const pathname = usePathname()
+
+  let color = 'blue'
+  if (payload && Object.keys(payload).includes(pathname.slice(1))) {
+    color = payload[pathname.slice(1)]
+  } 
+
+  return (
+    <button style={{ backgroundColor: color }}>
+      Click me! I am a useful button component
+    </button>
+  )
+}
+```
+
+### Cohort
+
+If instead, you want to target users who visited a page repeatedly, you can create and use a static cohort. For example, if you wanted to target a flag to show a new pricing page for people who have visited the pricing page repeatedly.
+
+1. Create a cohort for people who had multiple pageview events
+2. Duplicate as a static cohort.
+3. Use the static cohort as a condition for your feature flag
+
+> **Why can’t I use a behavioral cohort directly?** Calculating a behavioral (based on events) cohorts is slow. Feature flags care about speed. Adding behavioral cohorts to flag evaluation would significantly slow them down. Read [a full explanation in our docs](/docs/feature-flags/common-questions#why-cant-i-use-a-cohort-with-behavioral-filters-in-my-feature-flag).
+
+![Cohort creation video](../images/tutorials/group-page-machine-flags/cohort.mp4)
+
+## Targeting services, machines, devices, applications
+
+Another entity you might want to target is services, machines, or devices. For example, you might want to roll out a [canary release](/tutorials/canary-release) to some of your servers or activate a flag in one region but not another.
+
+### Custom user value
+
+A user for PostHog is just a distinct ID string connected to events. Although on the frontend, it is generated and connected to a real user's sessions, on the backend, it could mean anything. This includes service, machine, device, or application and we can use feature flags to work with them this way.
```

Comment:
```suggestion
A user in PostHog is just a distinct ID string connected to events. Although on the frontend, it is generated and connected to a real user's sessions, on the backend it could represent anything. This means you can use service, machine, device, or application IDs as the `distinct ID` – and feature flags will still work with them this way!
```